test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriBuilder/FilterAndOrderByBuilderTests.cs

```diff
@@ -465,6 +465,14 @@ public void BuildFilterWithDollarItInFunctionCall()
             Uri actualUri = UriBuilder(queryUri, ODataUrlKeyDelimiter.Parentheses, settings);
             Assert.Equal(new Uri("http://gobbledygook/People(1)/RelatedSSNs?$filter=endswith%28%24it%2C%27xyz%27%29"), actualUri);
         }
+
+        [Fact]
+        public void BuildFilterWithManyBinaryOperator()

```

Comment:
>BuildFilterWithManyBinaryOperator [](start = 20, length = 33)

Also, try changing the order of the operands:
Uri queryUri = new Uri("People?$filter=ID eq 1 or 2 eq ID or 3 eq 4 or ID eq 4 or5 eq ID", UriKind.Relative)